src/ring/adapter/jetty9.clj

```diff
@@ -44,21 +45,36 @@
     (string? response) {:body response}
     :else response))
 
+(defn- websocket-upgrade-response? [{:keys [status headers]}]
+  ;; HTTP 101 Switching Protocols
+  ;; https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/101
+  (and (= 101 status)
+       (let [headers (lower-case-keys headers)]
+         (and (= "websocket" (lower-case (get headers "upgrade")))
+              (= "upgrade" (lower-case (get headers "connection")))))))
+
 (defn ^:internal proxy-handler
   "Returns an Jetty Handler implementation for the given Ring handler."
   [handler]
-  (proxy [AbstractHandler] []
-    (handle [_ ^Request base-request ^HttpServletRequest request ^HttpServletResponse response]
-      (try
-        (let [request-map (build-request-map request)
-              response-map (-> (handler request-map)
-                               normalize-response)]
-          (when response-map
-            (servlet/update-servlet-response response response-map)))
-        (catch Throwable e
-          (.sendError response 500 (.getMessage e)))
-        (finally
-          (.setHandled base-request true))))))
+  (doto (ServletContextHandler.)
+    (.setContextPath "/*")
+    (.setAllowNullPathInfo true)
+    (JettyWebSocketServletContainerInitializer/configure nil)
```

Comment:
I wonder if this call is an expensive one? Perhaps we can move this to server start process?